gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyUtil.java

```diff
@@ -0,0 +1,309 @@
+import org.apache.spark.sql.SparkSession;
+import org.projectnessie.client.NessieClientBuilder;
+import org.projectnessie.client.StreamingUtil;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.client.http.HttpClientBuilder;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.LogResponse;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DistributedIdentifyUtil implements Serializable {
+
+  private static final long serialVersionUID = -5412153338307931076L;
+  private final SparkSession session;
+  private final Map<String, String> config;
+  private static final Logger LOGGER = LoggerFactory.getLogger(DistributedIdentifyUtil.class);
+
+  public DistributedIdentifyUtil(SparkSession session, Map<String, String> config) {
+    this.session = session;
+    this.config = config;
+  }
+
+  /**
+   * Builds the client builder; default({@link HttpClientBuilder}) or custom, based on the
+   * configuration provided.
+   *
+   * @param configuration map of client builder configurations.
+   * @return {@link NessieApiV1} object.
+   */
+  public static NessieApiV1 getApi(Map<String, String> configuration) {
+    String clientBuilderClassName = configuration.get("nessie.client-builder-impl");
+    if (clientBuilderClassName == null) {
+      // Use default HttpClientBuilder
+      return HttpClientBuilder.builder().fromConfig(configuration::get).build(NessieApiV1.class);
+    }
+    // Use custom client builder
+    NessieClientBuilder builder;
+    try {
+      builder =
+          (NessieClientBuilder)
+              Class.forName(clientBuilderClassName).getDeclaredConstructor().newInstance();
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException(
+          String.format("No custom client builder class found for '%s' ", clientBuilderClassName));
+    } catch (InvocationTargetException
+        | InstantiationException
+        | IllegalAccessException
+        | NoSuchMethodException e) {
+      throw new IllegalArgumentException(
+          String.format("Could not initialize '%s': ", clientBuilderClassName), e);
+    }
+    return (NessieApiV1) builder.fromConfig(configuration::get).build(NessieApiV1.class);
+  }
+
+  /**
+   * Compute the bloom filter per content id by walking all the live references in a distributed way
+   * using spark.
+   *
+   * @param references list of live references to walk.
+   * @param cutOffTimestampPerRef map of cutoff timestamp per reference.
+   * @param defaultCutOffTimestamp default cutoff timestamp to be used.
+   * @return map of {@link ContentBloomFilter} per content-id.
+   */
+  public Map<String, ContentBloomFilter> getLiveContentsBloomFilters(
```

Comment:
we don't have ContentId class and ContentId is string type, so returning string.

Do you mean contentKey? Each ref can call table in its own name. So, returning  ContentId instead of contentKey.